src/libraries/System.Private.Xml/src/System/Xml/Dom/DocumentSchemaValidator.cs

```diff
@@ -417,9 +417,8 @@ private void ValidateAttributes(XmlElement elementNode)
                 XmlQualifiedName attrQName;
                 for (int i = 0; i < _defaultAttributes.Count; i++)
                 {
-                    XmlSchemaAttribute schemaAttribute = (_defaultAttributes[i] as XmlSchemaAttribute)!;
+                    XmlSchemaAttribute schemaAttribute = (XmlSchemaAttribute)_defaultAttributes[i]!;
                     attrQName = schemaAttribute.QualifiedName;
-                    Debug.Assert(schemaAttribute != null);
```

Comment:
would it be better moving this one line up instead of deleting it?

Reply:
I don't have a strong opinion either way. there doesn't appear to be anything here that would really warrant the assert, so I'll leave it as is and avoid restarting ci